components/gfx/font_cache_thread.rs

```diff
@@ -460,7 +462,14 @@ impl FontCacheThread {
         self.chan.send(Command::GetFontTemplate(family, desc, response_chan))
             .expect("failed to send message to font cache thread");
 
-        let reply = response_port.recv()
+        let reply = response_port.recv();
+
+        if reply.is_err() {
+            let font_thread_has_closed = self.chan.send(Command::Ping).is_err();
+            assert!(font_thread_has_closed, "Failed to receive a response from live font cache");
+            panic!("Font cache thread has already exited.");
+        }
+        let reply = reply
             .expect("failed to receive response to font request");
```

Comment:
nit: this now fits in one line.

Reply:
Also, none of these need to use expect I guess, since we `panic!` on the line above... Want to either change it for `unwrap()`, or file an issue to revert this PR once we figure out the failures and reference it here?